storage-proofs/porep/src/nse/vanilla/labels.rs

```diff
@@ -214,7 +214,9 @@ fn mask_layer<D: Domain>(
         .par_chunks_mut(NODE_SIZE)
         .enumerate()
         .for_each(|(node_index, node)| {
-            let prefix = hash_prefix(LAYER_INDEX, node_index as u32, window_index);
+            let node_absolute_index =
+                window_index as u64 * config.num_nodes_window as u64 + node_index as u64;
+            let prefix = hash_prefix(LAYER_INDEX, node_absolute_index);
             let hash = Sha256::digest(&[&prefix[..], AsRef::<[u8]>::as_ref(replica_id)]);
```

Comment:
@dignifiedquire Is this a change to the actual encoding/sealing results? Changed tests below suggest it will be.

If that is correct, why is such a change introduced in the circuit implementation? Is this just an independent change/fix being  bundled here for convenience?

(cc: @keyvank in case the GPU implementation will need to change to agree with this)

Reply:
- yes this is a change to the actual hashing
- this is a bug fix
- it is bundled with this, as I realised this bug when trying to implement the circuit, and couldn't move forward without fixing it

Reply:
Okay, thanks. That's what I thought. @keyvank please take note that when this PR merges, the GPU implementation will need to be updated to synchronize results again.